src/schemapack/_internals/normalize.py

```diff
@@ -184,26 +190,40 @@ def _get_class_definition(
 
 
 def _should_embed(
-    embedding_profile: Mapping[str, Any] | None, relation_name: str
-) -> bool:
-    """Function to decide whether to embed a relation based on the embedding profile."""
-    if embedding_profile is None:
+    embedding_profile: EmbeddingProfile, relation_name: str
+) -> bool | None:
```

Comment:
Is the optional `None` necessary in the return type?
This will always return a boolean or raise